substrate-node/pallets/pallet-smart-contract/src/billing.rs

```diff
@@ -209,418 +342,420 @@ impl<T: Config> Pallet<T> {
             );
         }
 
-        // Finally update the lock
-        ContractLock::<T>::insert(contract.contract_id, &contract_lock);
-
-        log::info!("successfully billed contract with id {:?}", contract_id,);
+        contract_payment_state.last_updated_seconds = now;
+        contract_payment_state.cycles.defensive_saturating_inc();
+        ContractPaymentState::<T>::insert(contract.contract_id, &contract_payment_state);
 
         Ok(().into())
     }
 
+    fn ensure_contract_migrated(
+        account_id: &T::AccountId,
+        contract_lock: &types::ContractLock<BalanceOf<T>>,
+        contract_payment_state: &mut types::ContractPaymentState<BalanceOf<T>>,
+    ) {
+        if !contract_lock.is_migrated() {
```

Comment:
The `QueryKind` of the `ContractLock` storage map is `ValueQuery`, so you will always receive a default value if the key does not exist.
Since a contract lock is created and the 'lock_updated' field is updated with the current time every time a contract is created, this allows for easy distinction between a removed lock and an existing one.